### ISSUE TITLE
Polish instrument initialization and lookups

### ISSUE THREAD

**Issue by ktoso** [slashmo/gsoc-swift-tracing#86]
related question is how the `InstrumentationSystem` should be configured,
for instance should there be only one instrument, one of a kind and so on:

```swift
    // FIXME: smarter impl
    public static var tracer: TracingInstrument {
        self.lock.withReaderLock {
            let tracer: TracingInstrument? = self._tracer
            let res: TracingInstrument = tracer ?? NoOpTracingInstrument()
            return res
        }
    }
```

BTW the code above will not find a tracer if its provided by official and public `MultiplexInstrument`

_Originally posted by @pokryfka in https://github.com/slashmo/gsoc-swift-tracing/issues/31#issuecomment-662218351_

**Comment by ktoso** [comment]
Yeah the initialization and obtaining of tracers is not polished yet, can we not conflate the topics though?
New ticket for the fixing of initialization and tracer lookup would be useful.

I agree that it's not clear how we should deal with multiples if there's a tracer and non tracers yet; though that should be solvable IMO.

**Comment by slashmo** [comment]
@ktoso Can this be closed now as we already merged #101?

**Comment by ktoso** [comment]
Good point, I think so 👍
